File: dotnet/asyncdemo/asyncdemo.cs
using System;
using System.Threading.Tasks;

namespace asyncdemo
{
    class Program
    {
        async static Task Main(string[] args)
        {
            await SillyExample();
        }

        async static Task<int> ReturnNumberAsync(int n)
        {
            Console.WriteLine("In ReturnNumberAsync");
            return n;
        }

        async static Task SillyExample()
        {
            Console.WriteLine("Start of async function");
            var task = ReturnNumberAsync(10);
            Console.WriteLine("Called ReturnNumberAsync(10)");
            int n = await task;
            Console.WriteLine($"Result: {n}");
        }
    }
}

File: python/asyncdemo.py
#!/usr/bin/env python3

async def return_number_async(n):
    print('in return_number_async')
    return n

async def silly_example():
    print('start of async function')
    fut = return_number_async(10)
    print('called return_number_async(10)')
    n = await fut
    print(f'result: {n}')

if __name__ == '__main__':
    import asyncio
    asyncio.run(silly_example())

